libraries/core_libs/network/src/tarcap/packets_handlers/vote_packet_handler.cpp

```diff
@@ -20,6 +23,67 @@ void VotePacketHandler::validatePacketRlpFormat([[maybe_unused]] const PacketDat
   }
 }
 
+bool VotePacketHandler::shouldProcessVote(const std::shared_ptr<Vote> &vote, const std::shared_ptr<TaraxaPeer> &peer) {
+  const auto [current_pbft_round, current_pbft_period] = pbft_mgr_->getPbftRoundAndPeriod();
+  const auto current_pbft_step = pbft_mgr_->getPbftStep();
+  const auto vote_hash = vote->getHash();
+
+  if (vote->getPeriod() - 1 > pbft_chain_->getPbftChainSize() + kVoteAcceptingPeriods) {
```

Comment:
I did create comment before, which you resolved by yourself and then created the same thing. Do not use  getPbftChainSize here, use current_pbft_period